tests/contest/runtimetest/src/tests.rs

```diff
@@ -545,6 +545,139 @@ pub fn test_io_priority_class(spec: &Spec, io_priority_class: IOPriorityClass) {
     }
 }
 
+pub fn validate_memory_policy(spec: &Spec, expected_mode: Option<MemoryPolicyModeType>) {
+    if expected_mode.is_none() {
+        return;
+    }
+
+    let linux = spec.linux().as_ref().unwrap();
+    let memory_policy = linux.memory_policy();
+
+    let expected = expected_mode.unwrap();
+
+    if memory_policy.is_none() {
+        eprintln!("memory policy expected but not found in spec");
+        return;
+    }
+
+    let policy = memory_policy.as_ref().unwrap();
+    if policy.mode() != expected {
+        eprintln!(
+            "memory policy mode mismatch: expected {:?}, got {:?}",
+            expected,
+            policy.mode()
+        );
+        return;
+    }
+
+    // Read and parse /proc/self/numa_maps to verify the policy is applied
+    let numa_maps_content = match fs::read_to_string("/proc/self/numa_maps") {
+        Ok(content) => content,
+        Err(e) => {
+            eprintln!("failed to read /proc/self/numa_maps: {}", e);
+            return;
+        }
+    };
+
+    let first_line = match numa_maps_content.lines().next() {
+        Some(l) if !l.is_empty() => l,
+        _ => {
+            eprintln!("first line of /proc/self/numa_maps is empty");
+            return;
+        }
+    };
+
+    // Extract the policy part (usually the second field)
+    let parts: Vec<&str> = first_line.split_whitespace().collect();
+    if parts.len() < 2 {
+        eprintln!("unexpected format in /proc/self/numa_maps: {}", first_line);
+        return;
+    }
+
+    let policy_field = parts[1];
+
+    // Get auxiliary info from spec (nodes/flags)
+    let (nodes_is_empty, has_static_flag, has_relative_flag) = if let Some(p) = memory_policy {
+        let nodes_is_empty = p.nodes().as_ref().is_none_or(|n| n.trim().is_empty());
+        let mut has_static = false;
+        let mut has_relative = false;
+        if let Some(flags) = p.flags() {
+            for f in flags {
+                use oci_spec::runtime::MemoryPolicyFlagType::*;
+                match f {
+                    MpolFStaticNodes => has_static = true,
+                    MpolFRelativeNodes => has_relative = true,
+                    _ => {}
+                }
+            }
+        }
+        (nodes_is_empty, has_static, has_relative)
+    } else {
+        (true, false, false)
+    };
+
+    // Verify the policy matches expected mode
+    match expected_mode {
```

Comment:
Please ensure that, in the normal (successful) path, when nodes is specified, you verify that the allocation is on NUMA node 0; this information should be obtainable from the spec.